### PR TITLE
Decorator Class

### PR DESCRIPTION
**Pull request by AlucardSanin**
Check the usage of [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)

Creating Classes for new Methods and Decorators 
-- CapitalizeDecorator and TrimmerDecorator for Persons

### CODE REVIEW

File: Decorator/trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require_relative 'decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    if @nameable.correct_name.length > 10
+      "#{@nameable.correct_name[0..9]}.."
```
Comment (Kamba56) [inline Decorator/trimmer_decorator.rb]: - [ ] Great job trimming the name, but I notice you added `..` at the end of the name. Kindly remove this please, the name should only be trimmed.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @AlucardSanin, 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great use of Git workflow!👏👏👏
- [x] All linters passed
- [x] Great job adding decorators to your project.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- [OPTIONAL] Please in all future projects, kindly make sure your commit messages are meaningful and capitalized.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Decorator/trimmer_decorator.rb (hunk 1 above)
Comment (AlucardSanin) [reply Decorator/trimmer_decorator.rb]: Thanks for your review!
Microverse CR never stops surprising me, really blocked my approval for adding ".." to the end of the string.

**Review by alexander16108 — APPROVED** [review]
Hi @AlucardSanin , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ]  Great use of Git workflow!👏👏👏
- [ ]  All linters passed
- [ ]  Great job adding decorators to your project.


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Decorator/trimmer_decorator.rb (hunk 1 above)
Comment (Kamba56) [reply Decorator/trimmer_decorator.rb]: Also, you should know that microverse is training Micronuats to become industry professionals. Professionals pay attention to the most minute detail.

Comment (AlucardSanin) [reply Decorator/trimmer_decorator.rb]: Microverse tells Code Reviewvers to pay attention to project requirements, any other information you want to share should be marked as OPTIONAL, yes that badge that you probably received a lot during your journey as Micronaut but never used as Code Reviewver.

Comment (Kamba56) [reply Decorator/trimmer_decorator.rb]: I apologize sir. Just doing my job. :100: :)

Comment (Kamba56) [reply Decorator/trimmer_decorator.rb]: > Microverse tells Code Reviewvers to pay attention to project requirements, any other information you want to share should be marked as OPTIONAL, yes that badge that you probably received a lot during your journey as Micronaut but never used as Code Reviewver.

image
If you check the project requirements again, you will see that it says the name is to return a maximum of 10 characters. The extra `..` makes it 12 characters.

**Pull request merged**